modules/smart/main.nf

```diff
@@ -99,19 +104,26 @@ process SEARCH_SMART {
     tuple val(meta), path("hmmpfam.out"), path(fasta)
 
     script:
-    def commands = ""
-    smarts.each { smartFile ->
-        fasta.each { chunkFile ->
-            String hmmFilePath = "${dirpath.toString()}/${hmmdir}/${smartFile}.hmm"  // reassign to a var so the cmd can run
-            commands += "/opt/hmmer2/bin/hmmpfam"
-            commands += " --acc -A 0 -E 0.01 -Z 350000"
-            commands += " $hmmFilePath ${chunkFile} >> hmmpfam.out\n"
+    if (fasta.size() == 0) {
+        // Create an empty hmmpfam.out file if input is empty
+        """
+        touch hmmpfam.out
+        """
+    } else {
+        def commands = ""
+        smarts.each { smartFile ->
+            fasta.each { chunkFile ->
+                String hmmFilePath = "${dirpath.toString()}/${hmmdir}/${smartFile}.hmm"  // reassign to a var so the cmd can run
+                commands += "/opt/hmmer2/bin/hmmpfam"
+                commands += " --acc -A 0 -E 0.01 -Z 350000"
+                commands += " $hmmFilePath ${chunkFile} >> hmmpfam.out\n"
+            }
         }
```

Comment:
I think here it would be better for standardisation to stick with NF logic and the structure laid out in [PANTHER](https://github.com/ebi-pf-team/interproscan6/blob/0046d7b651a503d7e62561a807eda34e632cdcc1/modules/panther/main.nf#L119-L165) and [HAMAP](https://github.com/ebi-pf-team/interproscan6/blob/0046d7b651a503d7e62561a807eda34e632cdcc1/modules/hamap/main.nf#L79-L103), declare `commands` at the top then if/else statement to populate the variable then have the script string at the end.